cmd/promoted-image-governor/main.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	"github.com/openshift/ci-tools/pkg/config"
+	"github.com/openshift/ci-tools/pkg/steps/release"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/test-infra/prow/flagutil"
+	"os"
+	"path/filepath"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/util"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+type options struct {
+	ciOperatorconfigPath      string
+	kubeconfig                string
+	dryRun                    bool
+	ignoredImageStreamTagsRaw flagutil.Strings
+	ignoredImageStreamTags    sets.String
+}
+
+func newOpts() *options {
+	opts := &options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	// Controller-Runtimes root package imports the package that sets this flag
+	kubeconfigFlagDescription := "Path to the kubeconfig file to use for CLI requests."
+	if f := fs.Lookup("kubeconfig"); f != nil {
+		f.Usage = kubeconfigFlagDescription
+		// https://i.kym-cdn.com/entries/icons/original/000/018/012/this_is_fine.jpeg
+		defer func() { opts.kubeconfig = f.Value.String() }()
+	} else {
+		fs.StringVar(&opts.kubeconfig, "kubeconfig", "", kubeconfigFlagDescription)
+	}
+	fs.StringVar(&opts.ciOperatorconfigPath, "ci-operator-config-path", "", "Path to the ci operator config")
+	fs.BoolVar(&opts.dryRun, "dry-run", true, "Whether to run the controller-manager with dry-run")
+	fs.Var(&opts.ignoredImageStreamTagsRaw, "ignored-image-stream-tags", "An imagestreamtag to ignore while checking. It must be in namespace/name:tag format (e.G `ci/clonerefs:latest`). Can be passed multiple times.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("could not parse input")
+	}
+	return opts
+}
+
+func (o *options) validate() error {
+	if o.ciOperatorconfigPath == "" {
+		return fmt.Errorf("--ci-operator-config-path must be set")
+	}
+	return nil
+}
+
+func tagsToPromote(cfg *api.ReleaseBuildConfiguration) sets.String {
```

Comment:
If a function is used only once, it should not be a function unless there's a lot to be gained by giving it a name and hiding it

Reply:
Removed